src/middlewares/permissions.js

```diff
@@ -24,6 +27,9 @@ function createCheckPermission(environmentSecret, collectionName) {
     const renderingId = getRenderingIdFromUser(request.user);
     const permissionName = searchToEdit ? 'searchToEdit' : 'list';
 
+    // NOTICE: Unauthorize user if he's using a deprecated jwt
+    if (!renderingId) next(httpError(401, 'Deprecated JWT'));
+
```

Comment:
[Code] For maintenance, we shouldn't have code duplication between lines 12-14 and 30-32.

I don't know if you move the logic in it already, but isn't it the role of  `ensureAuthenticated` function in `auth.js` to respond 401 if the token has the old format.

You could implement something like:
```javascript
function ensureAuthenticated(request, response, next) {
  if (!request.user) {
    return next(new error.Unauthorized(ERROR_MESSAGE_DEFAULT));
  }
  
  // NOTICE: Automatically logout users trying to access the API with a token having an old data format.
  if (request.user.type) {
    return next(new error.Unauthorized(ERROR_MESSAGE_TOKEN_OLD));
  }
  
  return next();
}
```